components/butterfly/src/server/expire.rs

```diff
@@ -1,46 +1,50 @@
 //! Periodically check membership rumors to automatically "time out"
 //! `Suspect` rumors to `Confirmed`, and `Confirmed` rumors to
-//! `Departed`.
+//! `Departed`. Also purge any rumors that have expired.
 
-use crate::{rumor::{RumorKey,
-                    RumorType},
-            server::{timing::Timing,
-                     Server}};
+use crate::server::{timing::Timing,
+                    Server};
+use chrono::offset::Utc;
 use habitat_common::liveliness_checker;
 use std::{thread,
           time::Duration};
 
-const LOOP_DELAY_MS: u64 = 500;
+pub fn spawn_thread(name: String, mut server: Server, timing: Timing) -> std::io::Result<()> {
+    habitat_core::env_config_duration!(ExpireThreadSleepMillis, HAB_EXPIRE_THREAD_SLEEP_MS => from_millis, Duration::from_millis(500));
+    let sleep_ms: Duration = ExpireThreadSleepMillis::configured_value().into();
+
+    habitat_core::env_config_duration!(ExpireThreadPurgeSecs, HAB_EXPIRE_THREAD_PURGE_SECS => from_secs, Duration::from_secs(60));
+    let purge_secs: Duration = ExpireThreadPurgeSecs::configured_value().into();
 
-pub fn spawn_thread(name: String, server: Server, timing: Timing) -> std::io::Result<()> {
     thread::Builder::new().name(name)
-                          .spawn(move || -> ! { run_loop(&server, &timing) })
+                          .spawn(move || -> ! {
+                              run_loop(&mut server, &timing, sleep_ms, purge_secs)
+                          })
                           .map(|_| ())
 }
 
-fn run_loop(server: &Server, timing: &Timing) -> ! {
+fn run_loop(server: &mut Server, timing: &Timing, sleep_ms: Duration, purge_secs: Duration) -> ! {
```

Comment:
If I remember correctly, I made these parameters because every time you access the value of an env config variable, it prints a message to the log via `warn!` and that was drowning out all the other log messages.  I can double check this.